rules/dkarm_compat

```diff
@@ -0,0 +1,21 @@
+ifneq ($(strip $(DEVKITARM)),)
+
+include $(DEVKITARM)/base_rules
+
+else
+
+PREFIX		:=	arm-none-eabi-
+export CC	:=	$(PREFIX)gcc
+export CXX	:=	$(PREFIX)g++
+export AS	:=	$(PREFIX)as
+export AR	:=	$(PREFIX)gcc-ar
+export OBJCOPY	:=	$(PREFIX)objcopy
+export STRIP	:=	$(PREFIX)strip
+export NM	:=	$(PREFIX)gcc-nm
+export RANLIB	:=	$(PREFIX)gcc-ranlib
+
+endif
+
+ARCH := -march=armv4t -mtune=arm7tdmi -mthumb
+CFLAGS = $(ARCH) -Os -ffunction-sections -fdata-sections -fomit-frame-pointer -fno-inline -std=gnu11 -Wall
+LDFLAGS = $(ARCH) -nostartfiles -Wl,--nmagic,--gc-sections
```

Comment:
These 2 lines will break all modules and will make hekate a lot bigger and will not fit in the upper IRAM during injection. Having no -lgcc will not break the linking? LDFLAGS go through CC and not LD.

Also we use -O2 for all. Os is painfully slow.
In addition all modules use -marm for performance (since we don't care for size) and also we need -mthumb-interwork for swapping from thumb to arm for functions that need that performance boost.

Reply:
I'm using this and it works fine. Not sure why you think it would be larger - the resulting hekate.bin is actually smaller than the hekate build I've downloaded from someone else.

Adding back `-mthumb-interwork` is fine.